### PR TITLE
🔨 Improve typing 

### PR DESCRIPTION
**Pull request by robmorieson**
## What does this PR introduce?
- Fix type issues:
  - Label `HelpScout` prop types as optional 
  - Remove the `providerKey` type from `Messenger` props 
  - Add optional `locale` type to `LiveChatLoader` props
  - Set correct type for React children ([as per current recommendation](https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc#answer-71809927)) 
- Convert the example website to TS

Also some minor housekeeping 🧹 
- Align prettier options with other Calibre apps
- Add dedicated `.eslintrc` to the example website
- Rename the `static` directory to `public` in the example website ([static has been deprecated](https://nextjs.org/docs/messages/static-dir-deprecated))

### CODE REVIEW

File: src/components/LiveChatLoaderProvider.tsx [hunk 1]
```diff
@@ -5,11 +5,12 @@ import { LiveChatLoaderContext } from '../context'
 
 interface LiveChatLoaderProps {
   provider: Provider
-  children: JSX.Element
+  children: React.ReactNode
   idlePeriod?: number
   providerKey: string
   appID?: string
   baseUrl?: string
+  locale?: string
```
Comment (benschwarz) [inline src/components/LiveChatLoaderProvider.tsx:13]: Is `locale` is used by all implementations?

**Review by benschwarz — COMMENTED** [review]
All seems about right to me. Should there be a `README.md` update for props updates?

File: src/components/LiveChatLoaderProvider.tsx (hunk 1 above)
Comment (robmorieson) [reply src/components/LiveChatLoaderProvider.tsx:13]: Only **chatwoot** and **messenger** currently use `locale` - but in both instances, it's required during initialisation, hence why it's been implemented at the global `LiveChatLoaderProvider` level

Comment (benschwarz) [reply src/components/LiveChatLoaderProvider.tsx:13]: Seems worthy of a comment

Comment (robmorieson) [reply src/components/LiveChatLoaderProvider.tsx:14]: Good call. Done ✅

**Review by benschwarz — APPROVED** [review]
All good with me!

**Pull request merged**